src/apps/backend/modules/config/internals/config_file.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import annotations
+
+import os
+from pathlib import Path
+from typing import Any, Optional
+
+import yaml
+
+
+class ConfigFile:
+    """
+    Represents a configuration file that can be loaded, merged, and updated with environment variables.
+    """
+
+    def __init__(self, filename: str, config_path: Path):
+        self.filename = filename
+        self.config_path = config_path
+        self.content: dict[str, Any] = {}
+
+    def load(self) -> None:
+        """
+        Loads the content of the configuration file.
+        """
+        file_path = self.config_path / self.filename
+        try:
+            with open(file_path, "r", encoding="utf-8") as file:
+                self.content = yaml.safe_load(file) or {}
+        except FileNotFoundError:
+            from modules.logger.logger import Logger
+
+            Logger.error(message=f"Config file '{self.filename}' not found.")
+
+    def merge(self, other: ConfigFile) -> None:
```

Comment:
As a best practice, `merge` operator should return the merged result, and the object `self` should not be mutable.